src/main/java/com/amihaiemil/camel/YamlSerializer.java

```diff
@@ -0,0 +1,39 @@
+package com.amihaiemil.camel;
+
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * A Yaml serializer.
+ * @author sherif
+ * @version $Id$
+ * @since 1.0.0
+ */
+public interface YamlSerializer {
+    
+    /**
+     * Serialize a Map<Object, Object> to YamlNode.
+     * @param map Map<Object, Object>
+     * @return Yaml node
+     * @TODO #30 Implement the serialize method with Map input and unit-test it.
+     */
+    YamlNode serialize(final Map<Object, Object> map);
```

Comment:
@SherifWaly The interface should have only one method: ``YamlNode serialize();``. There will be 3 implementors of this interface. One implementor class will encapsulate a Map, another will encapsulate a Collection and the 3rd will encapsulate an Object.

e.g.
```java
YamlSerializer objectToYaml = YamlObjectSerializer(object);
YamlNode mapping = objectToYaml.serialize();

YamlSerializer mapToYaml = YamlMapSerializer(map);
YamlNode mapping = mapToYaml .serialize();

YamlSerializer collectionToYaml = YamlCollectionSerializer(collection);
YamlNode sequence= collectionToYaml .serialize();
```
Also, each implementor will override the return type of the method (you can do that, as long as the overriding type is a ``subtype`` of the initial one). E.g. ``YamlCollectionSerializer`` will have the method ``YamlSequence serialize();`` - that is possible because ``YamlSequence`` extends ``YamlNode``

The way you wrote it also a way of doing it, of course, but a class with these 3 methods would just be a bag of functions, I don't think it would encapsulate anything. An object should always encapsulate something - if it just has some utility methods, it is not a proper object, that's what I think.